pkg/router/template/plugin.go

```diff
@@ -291,6 +292,18 @@ func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, lookupSvc
 
 	out := make([]Endpoint, 0, len(endpoints.Subsets)*4)
 
+	// If addr is not a valid textual representation of an IP
+	// address, return addr. If it is valid and is IPv6 then
+	// format as "[" addr "]". The expectations in this function
+	// for addr are that it does not have a trailing :portnumber.
+	formatIPAddr := func(addr string) string {
+		ip := net.ParseIP(addr)
+		if ip != nil && strings.Count(addr, ":") >= 2 {
+			return "[" + addr + "]"
```

Comment:
Would it be better to use `ip` in this return line than `addr`? Also, I believe a valid ipv6 addr parsed by `net.ParseIP(addr)` will meet the condition `strings.Count(addr, ":") >= 2`. Thoughts?

Reply:
Edge cases: 

> Would it be better to use `ip` in this return line than `addr`? Also, I believe a valid ipv6 addr parsed by `net.ParseIP(addr)` will meet the condition `strings.Count(addr, ":") >= 2`. Thoughts?

This needs more thought and unit tests:

```go
package main

import (
	"fmt"
	"net"
)

func main() {
	ip := net.ParseIP("::FFFF:127.0.0.1")
	fmt.Println(ip)
	fmt.Println("is IPv4", ip.To4())
}
```

https://play.golang.org/p/4xXQTQNs45g

Reply:
> Still expecting this to fail CI haproxy tests where the system:router cluster role does not have privileges to list/watch endpointslices.
> 
> Needs: [openshift/openshift-apiserver#125](https://github.com/openshift/openshift-apiserver/pull/125)

Known failures are: 

```
"[sig-network][Feature:Router] The HAProxy router converges when multiple routers are writing status [Suite:openshift/conformance/parallel]"
"[sig-network][Feature:Router] The HAProxy router should override the route host for overridden domains with a custom value [Suite:openshift/conformance/parallel]"
"[sig-network][Feature:Router] The HAProxy router should override the route host with a custom value [Suite:openshift/conformance/parallel]"
"[sig-network][Feature:Router] The HAProxy router should run even if it has no access to update status [Suite:openshift/conformance/parallel]"
"[sig-network][Feature:Router] The HAProxy router should serve a route that points to two services and respect weights [Suite:openshift/conformance/parallel]"
"[sig-network][Feature:Router] The HAProxy router should serve the correct routes when scoped to a single namespace and label set [Suite:openshift/conformance/parallel]"
```

Reply:
With latest changed all are now passing with the exception of: 

```
"[sig-network][Feature:Router] The HAProxy router should serve a route that points to two services and respect weights [Suite:openshift/conformance/parallel]"
```